tools/autograd/templates/TraceType.cpp

```diff
@@ -1,17 +1,19 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
```

Comment:
Do we want to put this macro more in the headers that were cleaned?

Reply:
> Do we want to put this macro more in the headers that were cleaned?

It's sufficient for the header to be included by a file that defines the macro. Defining it in the header itself is possible, but you need to be very careful that it doesn't leak outside of the header. Otherwise, you might break the build of any source files that includes it. So, you have to guard it like this:

```
#pragma push_macro("TORCH_ASSERT_ONLY_METHOD_OPERATORS")
#define TORCH_ASSERT_ONLY_METHOD_OPERATORS

// rest of file

#pragma pop_macro("TORCH_ASSERT_ONLY_METHOD_OPERATORS")
```

Only defining it in source files is probably the safer route.

>Wondering about what should be the general policy around the forbid macro though?

Ideally this should be gradually introduced to all `.cpp` and `.cu` files until all of PyTorch can be compiled incrementally. At that point I would like to see it defined in the build system so all new files use `/ops` by default. Although you could explicitly opt-out by `undef`ing the macro so `derivatives.yaml`, for example, can still assume all operators are visible.